cmd/appgw-ingress/main.go

```diff
@@ -88,15 +88,22 @@ func main() {
 	agicPod := k8sContext.GetAGICPod(env)
 	metricStore := metricstore.NewMetricStore(env)
 
+	// get the details from Azure Context
+	azContext, err := azure.NewAzContext(env.AzContextLocation)
+	if err != nil {
+		glog.Info("Unable to load Azure Context file:", env.AzContextLocation)
+	}
+
+	// adjust env variable
 	if env.AppGwName == "" {
 		env.AppGwName = env.ReleaseName
 	}
 
-	if infraSubID, infraResourceGp, err := k8sContext.GetInfrastructureResourceGroupID(); env.SubscriptionID == "" && err != nil {
-		env.SubscriptionID = string(infraSubID)
-		if env.ResourceGroupName == "" {
-			env.ResourceGroupName = string(infraResourceGp)
-		}
+	if azContext != nil && env.SubscriptionID == "" {
```

Comment:
It seems that we are providing the option to read Subscription from an environment variable, which could overwrite the local file?
This looks good. If we don't have a real scenario, which requires that - we could even remove the option to provide Subscr ID via an env var to simplify.

Reply:
yes, we can think about this.

Reply:
We may have to think about bare metal